### DISCUSSION TITLE
How to quickly check if a file has 1 or more lines of text?

### DISCUSSION THREAD

**Discussion by PiotrMP006** [community/community#60681] [Programming Help]
### Select Topic Area

Question

### Body

Hi

How to quickly check if a file has 1 or more lines of text?

I'm not talking about counting all lines of text, just a quick check if the file has 1 or more lines.

Please help me

**Comment by nirav-gajera** [comment]
-in python :

def has_lines(file_path):
    with open(file_path, 'r') as file:
        first_line = file.readline()
        return bool(first_line)

- Usage
file_path = 'path/to/file.txt'
if has_lines(file_path):
    print("The file has one or more lines.")
else:
    print("The file is empty or does not exist.")

**Comment by spenserblack** [comment]
Even if you don't need the exact number, just seeing that the number is greater than 0 should be all you need.
```shell
wc -l < foo
```
If you want to get specific:
```shell
if [ "$(wc -l < foo)" -gt 0 ]; then
    echo "foo has 1 or more lines"
fi
```

Without more info it's hard to answer this question, though. Do you just need to do this in general? In a specific programming language?

**Comment by Beargruug** [comment]
If you are using vim you can do something like this: 

```
local function checkLines(_, bufnr)
  return vim.api.nvim_buf_line_count(bufnr) > 1
end
```

**Comment by queenofcorgis** [comment]
Love all the answers for different languages. Thank you @spenserblack, @L9us, and @nirav-gajera for jumping in ❤️ .
